keyboards/ploopyco/mouse/post_rules.mk

```diff
@@ -0,0 +1,3 @@
+ifneq ($(strip $(OPT_ENCODER_TYPE)), custom)
```

Comment:
To be blunt, I'm not sure that this is the best way to handle this.  There are already two implementations for the optical encoders, and this is introducing a third. 

Also, this doesn't filter against anything, to ensure that it's a valid selection, either.

Reply:
Me neither, but I'm outta ideas for this.

Right now the only "check" is that if you set the value to something non-existent, it will error out during build.

Reply:
Unfortunately, this does seem to be probably the best way to handle this.

However, looking at the code, most of the ploopy code is very copy+paste.  It would be worth pulling all of it out, and abstracting as much of it as possible.  I have a branch that has a lot of this done, but with the encoder abstraction PR.   And that may be the better direction, in general.

Reply:
When that PR lands I can port mine over too.

The graycode interpretation part is simple, it's just the analog conversion part that is different between implementations.